### ISSUE TITLE
CogpGPVar - simplification?

### ISSUE THREAD

**Issue by mpekalski** [danielkorzekwa/bayes-scala-gp#8]
In https://github.com/danielkorzekwa/bayes-scala-gp/blob/fa85daf09c79a3cd1ff3a2e7061613bfe6a38228/src/main/scala/dk/gp/cogp/model/CogpGPVar.scala

You have three lines in which I have impression could be simplified by taking just variance(y) instead of 1/variance(y). Because
1) vInv will have only values on diagonal and all are positive
2) cholesky of vinv will give you sqrt of vInv diagonal (as A_A.t = vInv)
3) transposing cholesky will not change anything (as (A_A.t).t = (A.t*A), but A it all the time a diagonal matrix so that equation is an identity for it)
4) then taking invchol pretty much multiplies two matrices with the same values on the diagonal, and should result in  1/vInv

So basically on diagonal it should true that
**v == 1/sqrt(vInv)^2**

```
    val m = DenseVector.zeros[Double](z.rows)
    val vInv = 0.1 * (1.0 / (variance(y))) * DenseMatrix.eye[Double](z.rows)
    val v = invchol(cholesky(vInv).t)
```

Moreover, taking inverse of a matrix is something that you should avoid, if possible, because of the numerical properties of the algorithm. Combined with using 1/10 on a big scale (see. http://ta.twi.tudelft.nl/users/vuik/wi211/disasters.html) it may bring unexpected results. Unless you wanted to introduce some random disturbance here.

Of course, I might have missed something and everything here is crucial for the algorithm to work. :)

**Comment by danielkorzekwa** [comment]
Yes, you are right, in this case no point for any inverse or cholesky. Still inverting matrices with cholesky decomposition is crucial for numerical stability when working with Gaussian Processes:https://en.wikipedia.org/wiki/Invertible_matrix#Cholesky_decomposition

Thanks.
